.github/workflows/binary.yaml

```diff
@@ -59,14 +67,19 @@ jobs:
           subject-path: "result/bin/dune"
           show-summary: false
 
+      # TODO: remove the extra Dune file when the complete move to tar is done.
       - name: Extract artifact and attestation
         run: |
-          mkdir -p ~/build/
-          cp ${{ steps.certificate.outputs.bundle-path }} result/bin/dune ~/build/
+          mkdir -p ~/build/$ARCHIVE_DIR/
+          cp ${{ steps.certificate.outputs.bundle-path }} ~/build
+          cp result/bin/dune ~/build/$ARCHIVE_DIR
+          cp result/bin/dune ~/build/
+          tar -czvf ~/build/$ARCHIVE_TARGZ -C ~/build $ARCHIVE_DIR
```

Comment:
I would suggest splitting tar and gzip, that way you can specify `gzip -9` which I think is a good idea in this case.

Maybe it could also make sense to set the `tar` format to `pax` (as described [in this post about archive formats](https://serverfault.com/questions/250511/which-tar-file-format-should-i-use)) but it's well possible that the `tar` version used in the action does that by default already.

Reply:
On my side, too. The manual states:
> __posix__
> [...] This archive format will be the default format for future versions of GNU tar. 
It is an alias for __pax__. I will use this format to make sure it is long compatible :+1: